cookbook/bundles/tests.md

```diff
@@ -0,0 +1,255 @@
+    $request = Request::createFromGlobals();
+
+    // The application handles the Request
+    $kernel = new AppKernel('prod', false);
+    $response = $kernel->handle($request);
+
+    // Dump the Response's headers and print the Response's content
+    $response->send();
+
+In this example Symfony is used as an application that mimicks the HTTP protocol,
+which means that it is possible to write transversal tests (System Tests) by building
+"manually" a ``Request``, giving it to our application and then check the returned ``Response``.
+
+This article provides examples on how to write such tests with third party bundles.
+
+Creating a minimal AppKernel
+----------------------------
+
+Third party bundles cannot be run on their own, they need an application. Instead
+of creating a separate application, install the bundle in it and then check how things turn out,
+it is possible to create a minimalistic ``AppKernel`` inside the bundle, for testing
+and showcase purposes.
+
+If the third party bundle is named ``AcmeMyBundle``, the ``AppKernel`` could look like this:
+
+    <?php
+    // File: Tests/app/AppKernel.php
+
+    namespace Acme\MyBundle\Tests;
+
+    use Symfony\Component\HttpKernel\Kernel;
+    use Symfony\Component\Config\Loader\LoaderInterface;
+
+    class AppKernel extends Kernel
+    {
+        public function registerBundles()
+        {
+            return array(
+                new Symfony\Bundle\FrameworkBundle\FrameworkBundle(),
+                new Acme\MyBundle\AcmeMyBundle(),
+            );
+        }
+
+        public function registerContainerConfiguration(LoaderInterface $loader)
+        {
+            $loader->load(__DIR__.'/config.yml');
+        }
+    }
+
+.. note::
+
+    ``AppKernel`` adds a direct dependency on Symfony's FrameworkBundle, which can be
+    installed by running the following command: ``composer require --dev symfony/framework-bundle``.
+
+In its minimum state ``AppKernel`` requires only one configuration parameter, ``secret``:
+
+    # File: Tests/app/config.yml
+    framework:
+        secret: Th1s1sS3cr3t!
+
+Symfony applications generate cache and logs directories, which can be ignored by
+adding those lines to ``.gitignore``:
+
+    # File: .gitignore
+    /Tests/app/cache
+    /Tests/app/logs
+    /vendor
+    /composer.lock
+
+Running commands
+----------------
+
+If the third party bundle provides a command, it can be helpful to create a ``console``
+similar to the one that can be found in Symfony's Standard Edition:
+
+    <?php
+    // File: Tests/app/console.php
+
+    set_time_limit(0);
+
+    require_once __DIR__.'/autoload.php';
+
+    use Symfony\Bundle\FrameworkBundle\Console\Application;
+
+    $kernel = new AppKernel('dev', true);
+    $application = new Application($kernel);
+    $application->run();
+
+With this it becomes possible to run manually the command:
+
+    php Tests/app/console.php
```

Comment:
commands are prefixed with `$`. As this is not PHP code as well, it has to be preceded by `.. config-block:: bash`